server/storage/mvcc/kvstore_compaction.go

```diff
@@ -16,14 +16,23 @@ package mvcc
 
 import (
 	"encoding/binary"
+	"fmt"
 	"time"
 
 	"go.etcd.io/etcd/server/v3/storage/schema"
 	"go.uber.org/zap"
 )
 
-func (s *store) scheduleCompaction(compactMainRev int64, keep map[revision]struct{}) bool {
+func (s *store) scheduleCompaction(compactMainRev int64) (kvHash, error) {
+	s.revMu.RLock()
+	minHashRev := s.compactMainRev
+	s.revMu.RUnlock()
```

Comment:
No need to acquire the read lock, we can use 
```
minHashRev := atomic.LoadInt64(&s.compactMainRev)
```

Reply:
Here I wanted to follow the same locking mechanism used by `HashByRev` which uses `revMu`. From code:
```
	// revMu protects currentRev and compactMainRev.
	// Locked at end of write txn and released after write txn unlock lock.
	// Locked before locking read txn and released after locking
```

Reply:
Reverted this code this was incorrect.